Paintroid/src/main/java/org/catrobat/paintroid/dialog/IndeterminateProgressDialog.kt

```diff
@@ -18,18 +18,23 @@
  */
 package org.catrobat.paintroid.dialog
 
-import android.annotation.SuppressLint
-import android.content.Context
+import android.app.Dialog
+import android.os.Bundle
 import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
 import org.catrobat.paintroid.R
 
-object IndeterminateProgressDialog {
-    @JvmStatic
-    @SuppressLint("InflateParams")
-    fun newInstance(context: Context): AlertDialog {
-        return AlertDialog.Builder(context, R.style.PocketPaintProgressDialog)
-            .setCancelable(false)
+class IndeterminateProgressDialog : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        isCancelable = false
+        return AlertDialog.Builder(requireContext(), R.style.PocketPaintProgressDialog)
             .setView(R.layout.pocketpaint_layout_indeterminate)
             .create()
```

Comment:
```suggestion
        return AlertDialog.Builder(requireContext(), R.style.PocketPaintProgressDialog)
            .setView(R.layout.pocketpaint_layout_indeterminate)
            .setCancelable(false)
            .create()
```

Reply:
what do you think about this? You may then use expression-body syntax too ;D

Reply:
so I also tried this earlier but this unfortunately didn't work, because the isCancelable is related to the DialogFragment class and for some reasons doesn't work for the `Alert.Dialog.Builder()` (the dialog/dialogfragment was cancelable for some reasons)

Reply:
hmm.. ok 👍🏻